examples/gno.land/p/demo/md/md.gno

```diff
@@ -0,0 +1,54 @@
+package md
+
+import (
+	"strings"
+)
+
+// escape a URL for markdown.
+func EscMDURL(s string) string {
+	return escape(s)
+}
+
+func escape(s string) string {
+	schars := map[string]string{
+		"%":  "%25",
+		" ":  "%20",
+		"!":  "%21",
+		"\"": "%22",
+		"#":  "%23",
+		"$":  "%24",
+		"&":  "%26",
+		"'":  "%27",
+		"(":  "%28",
+		")":  "%29",
+		"*":  "%2A",
+		"+":  "%2B",
+		",":  "%2C",
+		":":  "%3A",
+		";":  "%3B",
+		"<":  "%3C",
+		"=":  "%3D",
+		">":  "%3E",
+		"?":  "%3F",
+		"@":  "%40",
+		"[":  "%5B",
+		"\\": "%5C",
+		"]":  "%5D",
+		"^":  "%5E",
+		"_":  "%5F",
+		"`":  "%60",
+		"{":  "%7B",
+		"|":  "%7C",
+		"}":  "%7D",
+		"~":  "%7E",
+		"\n": "%0A",
+	}
```

Comment:
I think this might be better being outside the function for performance reasons when executed on the Gno VM.